### PR TITLE
Search with google feature

### PR DESCRIPTION
**Pull request by Y0dax**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Added a basic search with google feature for selected text in channel messages. (#2058)

There is ongoing discussion regarding allowing different search engines, but I figured it would be good to get the basic feature out there which can be enhanced later.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Unversioned
 
+- Minor: Added search with google feature (#2058)
```
Comment (leon-richardt) [inline CHANGELOG.md:5]: Please place "Minor" entries in the changelog below the "Major" entries.

File: src/widgets/helper/ChannelView.cpp [hunk 2]
```diff
@@ -1706,11 +1706,18 @@ void ChannelView::addContextMenuItems(
...
     if (!this->selection_.isEmpty())
     {
         menu->addAction("Copy selection",
                         [this] { crossPlatformCopy(this->getSelectedText()); });
+
+        menu->addAction("Search with Google", [this] {
+            auto query =
+                QString(QUrl::toPercentEncoding(this->getSelectedText()));
+            QDesktopServices::openUrl(
+                QUrl("https://google.com/search?q=" + query));
```
Comment (leon-richardt) [inline src/widgets/helper/ChannelView.cpp:1719]: I do not like this string being hardcoded here. Consider extracting the opening of the link into an own class, especially in regards to custom search query schemas potentially being added later on.

**Review by leon-richardt — CHANGES_REQUESTED** [review]
Important note: Should probably wait for @fourtf or @pajlada to express their opinion on #2058 before investing too much time into implementation.

**Comment by Y0dax** [comment]
Closing while discussion is ongoing.

**Pull request closed without merging**